src/Microsoft.DotNet.Build.Tasks/Targets/publishtest.targets

```diff
@@ -41,6 +42,14 @@
     <ItemGroup>
       <TestCopyLocal Include="@(RunTestsForProjectInputs)" Exclude="@(PackagesConfigs)" />
     </ItemGroup>
+    
+    <ItemGroup Condition="'$(TestWithLocalLibraries)'=='true'">
+      <!-- Replace some of the resolved libraries that came from nuget by exploring the list of files that we are going to copy
+           and replacing them with local copies that were just built -->
+      <TestCopyLocalReplaceCandidates Include="@(TestCopyLocal -> '$(BaseOutputPath)$(OSPlatformConfig)\%(filename)\%(filename).dll')" />
+      <TestCopyLocalReplaceCandidates Include="@(TestCopyLocal -> '$(BaseOutputPath)$(OSPlatformConfig)\%(filename)\%(filename).pdb')" />
+      <TestCopyLocal Include="@(TestCopyLocalReplaceCandidates)" Condition="Exists('%(TestCopyLocalReplaceCandidates.FullPath)')" />
```

Comment:
Given that you are putting these later in the TestCopyLocal list that will likely cause the file to be copied twice. Can you put these in a separate item group and put that at the front of the copy command?